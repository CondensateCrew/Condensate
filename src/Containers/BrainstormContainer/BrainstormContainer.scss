@import "../../css/variables.scss";

.bs-container {
  // break-inside: avoid;
  column-count: 2;
  column-width: 28rem;
  column-gap: $medium-space;
  margin-top: $medium-space;
  margin-left: $medium-space;
  padding-right: $medium-space;
}
