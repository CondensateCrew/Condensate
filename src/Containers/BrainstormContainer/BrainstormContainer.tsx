import './BrainstormContainer.scss';
import React from 'react';
import { useSelector } from "react-redux";
import { Brainstorm, AppStore } from 'interfaces';
import BrainstormCard from 'Components/BrainstormCard/BrainstormCard';

const BrainstormContainer:React.FC = () => {
  const { allBrainstorms, queryValue, filterValue} = useSelector((state: AppStore) => ({
    allBrainstorms: state.allBrainstorms,
    queryValue: state.query,
    filterValue: state.filter
  }));

  const filteredBrainstorms: (Brainstorm | undefined)[] = allBrainstorms
    .filter((bs: Brainstorm) => (
      filterValue === 'genius' && bs.isGenius === true  //eslint-disable-line
      || //eslint-disable-line
      bs.categories.find(ctg => {
        return (filterValue === '')
          ? ctg.name !== filterValue
          : ctg.name === filterValue
      })
      && (bs.question.toLowerCase().includes(queryValue.toLowerCase()) //eslint-disable-line
      || bs.response.toLowerCase().includes(queryValue.toLowerCase())
      || bs.action.action.toLowerCase().includes(queryValue.toLowerCase()))
    ));

  const cards:JSX.Element[] = filteredBrainstorms.map((bs:(Brainstorm | undefined) ):any => {
    if (bs) {
      return <BrainstormCard key={`bs-${bs.id}`} {...bs} />
    }
    return undefined
  });

  return (
    <section className="bs-container">
      { cards }
    </section>
  )
}

export default BrainstormContainer;
