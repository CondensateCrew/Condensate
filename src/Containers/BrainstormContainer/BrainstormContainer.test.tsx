import React from 'react';
import BrainstormContainer from './BrainstormContainer';
import mockBrainstorms from './mockData'
import { shallow } from 'enzyme';
import { useSelector, useDispatch } from "react-redux";

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: () => ({
    allBrainstorms: mockBrainstorms,
    queryValue: '',
    filterValue: ''
  }),
  useDispatch: () => mockDispatch
}));

describe('BrainstormContainer component', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<BrainstormContainer />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
