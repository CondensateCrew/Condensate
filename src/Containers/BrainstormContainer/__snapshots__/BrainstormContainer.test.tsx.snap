// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrainstormContainer component should match the snapshot 1`] = `
<section
  className="bs-container"
>
  <BrainstormCard
    action={
      Object {
        "action": "Cook breakfast",
        "id": 1,
      }
    }
    categories={
      Array [
        Object {
          "id": 1,
          "name": "Food",
        },
      ]
    }
    id={1}
    isGenius={true}
    key="bs-1"
    question="What can I eat for breakfast?"
    response="A good breakfast should combine protein, stick-to-your-ribs fat, and some fiber-filled carbs. This is your key to feeling satisfied, energized, and not ready to eat your stapler by lunch. To make any of the breakfast combos suggested below heartier, up the veggies and/or fruit — these are unlimited as far as I’m concerned, especially at breakfast."
  />
  <BrainstormCard
    action={
      Object {
        "action": "Build an application",
        "id": 2,
      }
    }
    categories={
      Array [
        Object {
          "id": 2,
          "name": "Tech",
        },
      ]
    }
    id={2}
    isGenius={false}
    key="bs-2"
    question="What app can I create with React?"
    response="Aside from products, e-commerce apps may provide a service to consumers. If it is a service that is provided locally, an interactive map could be added to the app to provide for the service provider and customer to know each others' location. Food delivery applications such as UberEats and Doordash, which require the location of the person ordering the food, come to mind."
  />
  <BrainstormCard
    action={
      Object {
        "action": "Make money",
        "id": 3,
      }
    }
    categories={
      Array [
        Object {
          "id": 3,
          "name": "Money",
        },
      ]
    }
    id={3}
    isGenius={false}
    key="bs-3"
    question="How to make money fast?"
    response="People are always looking to have their cars washed and detailed. You could be a mobile car washer and detailer without having a permanent location. Reach out to people you know or make some flyers and put it in your neighbors' mailboxes. If you want to get serious about it, prop up a one-page website or give out business cards. You can make money quickly doing this."
  />
</section>
`;
